gateway-server/src/test/java/org/apache/knox/gateway/util/DescriptorGeneratorTest.java

```diff
@@ -0,0 +1,55 @@
+package org.apache.knox.gateway.util;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.File;
+import java.nio.charset.Charset;
+import java.util.Arrays;
+import java.util.List;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.knox.gateway.model.DescriptorConfiguration;
+import org.apache.knox.gateway.model.Topology;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class DescriptorGeneratorTest {
+  private static final ObjectMapper mapper = new ObjectMapper();
+  private static final String TEST_DESC_1 = "test_desc1.json";
+  private static final String TEST_PROV_1 = "test_prov1.json";
+  private static final String IMPALA_UI = "IMPALAUI";
+  private static final List<String> URLS =
+          Arrays.asList("http://amagyar-1.test.site:25000/", "http://amagyar-2.test.site:25000");
+  @Rule
+  public TemporaryFolder folder= new TemporaryFolder();
```

Comment:
Instead of making this a temporary file, we could create a static file in `src/test/resources` and have the code read that file. Since this is not a dynamic one, I think it would make more sense and we would avoid creating a temporary folder every time this test is running (the read I/O operation is unavoidable, but creation can be).

Reply:
This is not for creating an input file, but for the resulting output file. The generrated descriptor will be placed into this temp folder (and junit will clean it automatically).